genesyscloud/conversations_messaging_settings/resource_conversations_messaging_settings.go

```diff
@@ -109,6 +109,17 @@ func deleteConversationsMessagingSettings(ctx context.Context, d *schema.Resourc
 	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
 	proxy := getConversationsMessagingSettingsProxy(sdkConfig)
 
+	// If a messaging setting is set as a default config for an organization, the API will throw a 400 causing tests to fail
+	// Check that messaging setting is not the default setting before deletion
+	defaultSetting, response, getErr := proxy.getConversationsMessagingSettingsDefaultId(ctx)
+	if getErr != nil {
+		return util.BuildAPIDiagnosticError(resourceName, fmt.Sprintf("Failed to get conversations messaging setting default %s error: %s", d.Id(), getErr), response)
+	}
+	if defaultSetting != nil && *defaultSetting.Id == d.Id() {
+		log.Printf("Messaging Settings: %s cannot be deleted since it is the default config for an organization", d.Id())
+		return nil
+	}
```

Comment:
Realised by the end of this review that they don't return 404s if it doesn't exists. Might still be worth implementing something along those lines in case they ever fix up their API in the future